src/main/java/com/aws/greengrass/shadowmanager/configuration/RateLimitsConfiguration.java

```diff
@@ -0,0 +1,126 @@
+    @Getter
+    private final int maxLocalRequestRatePerThing;
+    @Getter
+    private final int maxOutboundUpdatesPerSecond;
+    private final InboundRateLimiter inboundRateLimiter;
+    private final IotDataPlaneClientWrapper iotDataPlaneClientWrapper;
+
+    private static int getMaxTotalLocalRequestRateFromTopics(Topics rateLimitsTopics) {
+        int maxTotalLocalRequestRate = Coerce.toInt(rateLimitsTopics
+                .lookup(CONFIGURATION_MAX_TOTAL_LOCAL_REQUESTS_RATE)
+                .dflt(DEFAULT_TOTAL_LOCAL_REQUESTS_RATE));
+        Validator.validateTotalLocalRequestRate(maxTotalLocalRequestRate);
+        return maxTotalLocalRequestRate;
+    }
+
+    private static int getMaxLocalRequestRatePerThingFromTopics(Topics rateLimitsTopics) {
+        int maxLocalRequestRatePerThing = Coerce.toInt(rateLimitsTopics
+                .lookup(CONFIGURATION_MAX_LOCAL_REQUESTS_RATE_PER_THING_TOPIC)
+                .dflt(DEFAULT_LOCAL_REQUESTS_RATE));
+        Validator.validateLocalShadowRequestsPerThingPerSecond(maxLocalRequestRatePerThing);
+        return maxLocalRequestRatePerThing;
+    }
+
+    private static int getMaxOutboundUpdatesPerSecondFromTopics(Topics rateLimitsTopics) {
+        int maxOutboundUpdatesPerSecond = Coerce.toInt(rateLimitsTopics
+                .lookup(CONFIGURATION_MAX_OUTBOUND_UPDATES_PS_TOPIC)
+                .dflt(DEFAULT_MAX_OUTBOUND_SYNC_UPDATES_PS));
+        Validator.validateOutboundSyncUpdatesPerSecond(maxOutboundUpdatesPerSecond);
+        return maxOutboundUpdatesPerSecond;
+    }
+
+    private RateLimitsConfiguration(int maxLocalRequestRatePerThing,
+                                    int maxTotalLocalRequestRate,
+                                    int maxOutboundUpdatesPerSecond,
+                                    InboundRateLimiter inboundRateLimiter,
+                                    IotDataPlaneClientWrapper iotDataPlaneClientWrapper) {
+
+        this.maxLocalRequestRatePerThing = maxLocalRequestRatePerThing;
+        this.maxTotalLocalRequestRate = maxTotalLocalRequestRate;
+        this.maxOutboundUpdatesPerSecond = maxOutboundUpdatesPerSecond;
+        this.inboundRateLimiter = inboundRateLimiter;
+        this.iotDataPlaneClientWrapper = iotDataPlaneClientWrapper;
+    }
+
+    private static RateLimitsConfiguration getRateLimitsConfigurationFromTopics(Topics topics) {
+        Topics rateLimitsTopics = topics.lookupTopics(CONFIGURATION_RATE_LIMITS_TOPIC);
+        int maxTotalLocalRequestRate = getMaxTotalLocalRequestRateFromTopics(rateLimitsTopics);
+        int maxLocalRequestRatePerThing = getMaxLocalRequestRatePerThingFromTopics(rateLimitsTopics);
+        int maxOutboundUpdatesPerSecond = getMaxOutboundUpdatesPerSecondFromTopics(rateLimitsTopics);
+        InboundRateLimiter inboundRateLimiter = topics.getContext().get(InboundRateLimiter.class);
+        IotDataPlaneClientWrapper iotDataPlaneClientWrapper = topics.getContext().get(IotDataPlaneClientWrapper.class);
```

Comment:
Seems weird that the configuration object depends on a `Client`. Seems like this dependencies should be inverted, in fact both `InboundRateLimiter` and the `InboundRateLimiter` depend on the configuration not the other way around. If we relied on events that would be an easier change but for now what we could do is:

1. Move the `InboundRateLimiter` and `IotDataPlaneClientWrapper` update logic to [here](IotDataPlaneClientWrapper)

2. Check if the configuration for the rate limited changed by

```
           componentConfiguration = ComponentConfiguration.from(componentConfiguration, getConfig());
           rateLimitConfig = componentConfiguration.getRateLimitConfiguration();

          if (rateLimitConfig.hasChanged(componentConfiguration.getRateLimitConfig()) {
               // Crate a method here to update those classes or just inline update them here
          }

```

Reply:
Updated the PR by moving these back in main ShadowManager file.